File: Web Development/Front-End/JavaScript/BubbleSort.js
document.getElementsByClassName('btn-input')[0].addEventListener('click',updateArray)
var arr=new Array()
function updateArray(event){

    var button=event.target
    var number=button.parentElement.getElementsByClassName('number-input')[0].value
    arr.push(parseInt(number))
    // console.log(button,number)
    document.getElementsByClassName('array-input-list')[0].innerHTML+=`<span>${number} </span>`
}
document.getElementsByClassName('btn-output')[0].addEventListener('click',bubbleSort)
function bubbleSort(){
    for(var i=0;i<arr.length-1;i++){
        for(var j=0;j<arr.length-i-1;j++){
            if(arr[j]>arr[j+1]){
                var temp=arr[j]
                arr[j]=arr[j+1]
                arr[j+1]=temp
            }
        }
    }
    sortedArray(arr)
}
function sortedArray(arr){
    document.getElementsByClassName('array-output-list')[0].innerHTML=`<span></span>`
    for(var i=0;i<arr.length;i++){
        var number=arr[i]
        document.getElementsByClassName('array-output-list')[0].innerHTML+=`<span>${number} </span>`
    }
}

File: DSA/Recursion/BubbleSort.c
#include<stdio.h>

void swap(int*a,int*b){
	int temp=*a;
	*a=*b;
	*b=temp;
}
void BubbleSort(int arr[],int size){
	if(size==0||size==1)return;
	for(int i=0;i<size-1;i++){
		if(arr[i]>arr[i+1]){
			swap(&arr[i],&arr[i+1]);
		}
	}
	BubbleSort(arr,size-1);
}
void printArray(int arr[],int size){
	printf("Array :  ");
	for(int i=0;i<size;i++){
		printf("%d ",arr[i]);
	}
	printf("\n");
}
void main(){
	int arr[]={2,6,4,1,8,0};
	int size=sizeof(arr)/sizeof(int);
	printArray(arr,size);
	BubbleSort(arr,size);
	printArray(arr,size);
}